### PR TITLE
Add Finite-Quiz test plans

### PR DESCRIPTION
**Pull request by marlau97**
Closes Gamify-IT/issues#246

### CODE REVIEW

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Version: V1.0, 2022-08-26
+Author: Martin Lautenschlager, Levi Otterbach
+
+## Description
+
+This use case verifies that a normal run through the game works correctly.
+
+## Precondition
+
+A configuration has been chosen that both the frontend and the backend know about. The backend has stored test data about this configuration.
```
Comment (maexled) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:12]:
```suggestion
A configuration has been chosen that exists in the backend. At least two questions exist in the given configuration.
```
Why backend and **frontend** know about it? I mean just backend needs to know and frontend receives a response.
Also in this test case, the configuration needs at least 2 questions.

Comment (delvh) [reply dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:12]: I think what they meant was "the frontend knows which configuration to send".

**Comment by delvh** [comment]
By the way, the docker-compose should be adapted in the repo.
I just wanted to test it, only to see that it still points at chickenshock.

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+3. The tester chooses an answer by clicking the button of the answer they want to choose.
+4. After answering, corresponding feedback wether the question was answered right or wrong is shown for a short time
+5. After the feedback, the next question is loaded
+6. Once the tester answered all questions, a feedback screen is shown that informs the tester of his overall performance
+7. Pressing "Close" exits the game
+
+## Alternative procedures
+
+x.1. Clicking "Restart" restarts the game with the same config \
+x.2. Clicking "Close" exits the game
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:32]:
```suggestion
5.1. There is no next question - skip to step `6` \
7.1. Clicking `Restart` restarts the game with the same config
```

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+A configuration has been chosen that both the frontend and the backend know about. The backend has stored test data about this configuration.
+
+## Postcondition
+
+The game result the tester achieved is saved in the finite-quiz database.
+The game shows the achieved result.
+
+## Typical procedure
+
+1. The game is loaded with a valid config
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:21]:
```suggestion
1. The game has been started
```

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+2. A question with corresponding answers of the given config is loaded. The question is shown in the big box in the upper half of the screen and there are as many buttons in the lower half of the screen as there are possible answers.
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:22]:
```suggestion
2. A question and the possible answers are loaded. Both the question and all possible answers are presented visually as boxes. The question is visually distinct from its answer options
```

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+3. The tester chooses an answer by clicking the button of the answer they want to choose.
+4. After answering, corresponding feedback wether the question was answered right or wrong is shown for a short time
+5. After the feedback, the next question is loaded
+6. Once the tester answered all questions, a feedback screen is shown that informs the tester of his overall performance
+7. Pressing "Close" exits the game
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:27]:
```suggestion
7. Pressing `Close` exits the game
```

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+## Postcondition
+
+The game result the tester achieved is saved in the finite-quiz database.
+The game shows the achieved result.
+
+## Typical procedure
+
+1. The game is loaded with a valid config
+2. A question with corresponding answers of the given config is loaded. The question is shown in the big box in the upper half of the screen and there are as many buttons in the lower half of the screen as there are possible answers.
+3. The tester chooses an answer by clicking the button of the answer they want to choose.
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:23]:
```suggestion
3. The tester chooses an answer by clicking on the answer they want to choose
```

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+4. After answering, corresponding feedback wether the question was answered right or wrong is shown for a short time
+5. After the feedback, the next question is loaded
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:25]:
```suggestion
4. For a short time, feedback is shown wether the chosen answer is correct or incorrect
5. The next question is loaded
```

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md (hunk 2 above)
Comment (delvh) [reply dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:32]: Also, the restart button should probably be disabled while the game is active…

Comment (marlau97) [reply dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:32]: I don't think so, if you get a question wrong and want to aim for 100% you would have to play the whole game before you can try again, that would be annoying

Comment (delvh) [reply dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md]: Yes, that's how every single quiz in history behaves?
Tell me one game that let's you immediately restart?

File: dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+The game result the tester achieved is saved in the finite-quiz database.
+The game shows the achieved result.
+
+## Typical procedure
+
+1. The game has been started
+2. A question and the possible answers are loaded. Both the question and all possible answers are presented visually as boxes. The question is visually distinct from its answer options
+3. The tester chooses an answer by clicking on the answer they want to choose
+4. For a short time, feedback is shown wether the chosen answer is correct or incorrect
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/finite-quiz/u-finite-quiz-1-playthrough.md:24]:
```suggestion
4. For a short time, feedback is shown whether the chosen answer is correct or incorrect
```

**Review by delvh — APPROVED** [review]

**Pull request merged**